src/track/taglib/trackmetadata_id3v2.cpp

```diff
@@ -142,29 +190,26 @@ TagLib::ID3v2::CommentsFrame* findFirstCommentsFrame(
         const QString& description,
         bool preferNotEmpty = true) {
     TagLib::ID3v2::CommentsFrame* pFirstFrame = nullptr;
-    // Bind the const-ref result to avoid a local copy
-    const TagLib::ID3v2::FrameList& commentsFrames =
-            tag.frameListMap()["COMM"];
-    for (TagLib::ID3v2::FrameList::ConstIterator it(commentsFrames.begin());
-            it != commentsFrames.end();
-            ++it) {
-        auto* pFrame =
-                dynamic_cast<TagLib::ID3v2::CommentsFrame*>(*it);
-        if (pFrame) {
-            const QString frameDescription(
-                    toQString(pFrame->description()));
-            if (0 == frameDescription.compare(description, Qt::CaseInsensitive)) {
-                if (preferNotEmpty && pFrame->toString().isEmpty()) {
-                    // we might need the first matching frame later
-                    // even if it is empty
-                    if (!pFirstFrame) {
-                        pFirstFrame = pFrame;
-                    }
-                } else {
-                    // found what we are looking for
-                    return pFrame;
-                }
+    for (TagLib::ID3v2::Frame* pFrame : tag.frameListMap()["COMM"]) {
+        DEBUG_ASSERT(pFrame);
+        auto* pNextFrame = downcastFrame<TagLib::ID3v2::CommentsFrame>(pFrame);
```

Comment:
constness of pFrame got lost. pNextFrame could also be const, if I'm not mistaken.

Reply:
https://github.com/mixxxdj/mixxx/pull/4610/commits/e41ddf669293ea034abc2c10f28bc570b650fe32

Adding `const` qualifiers wherever suitable in C++ is controversial. It adds a lot of noise and for function parameters it even changes the function's signature. Fortunately, this is not the case for local loop variables, but rather uncommon.

Use Rust and never ever think about it again.

Reply:
Our coding guideline results in a mixture between *west const* and *east const* placement adds even more confusion to pointer variables. Therefore I always was an advocate of using *east const* consistently that could prevent a lot of misinterpretation about to what the `const` keyword is actually applied, i.e. everything left to the keyword for *east const*. It could be that simple.

Reply:
https://hackingcpp.com/cpp/design/east_vs_west_const.html